mllib/src/main/scala/org/apache/spark/mllib/stat/test/ChiSqTest.scala

```diff
@@ -109,9 +109,10 @@ private[stat] object ChiSqTest extends Logging {
           }
           i += 1
           distinctLabels += label
-          features.toArray.view.zipWithIndex.slice(startCol, endCol).map { case (feature, col) =>
-            allDistinctFeatures(col) += feature
-            (col, feature, label)
+          features.toArray.slice(startCol, endCol).zip(startCol until endCol).map {
```

Comment:
Thanks @squito for the great suggestion. I would wait for a while to see if there're other comments.